### ISSUE TITLE
_init method that has parameters shows non-descriptive error

### ISSUE THREAD

**Issue by PLyczkowski** [godotengine/godot#15866]
**Godot version:**
![image](https://user-images.githubusercontent.com/7307968/35147008-49fa8dda-fd0d-11e7-82e9-3f14f1b1bf93.png)

**Issue description:**
When a script with an _init method that uses parameters is added to a node in a scene, the scene fails to load and the error shown is very non-descriptive:
![image](https://user-images.githubusercontent.com/7307968/35146946-001c9c62-fd0d-11e7-8f30-06d2eab1e2c8.png)

**Minimal reproduction project:**
[Init Error Godot.zip](https://github.com/godotengine/godot/files/1646195/Init.Error.Godot.zip)

**Comment by akien-mga** [comment]
Indeed, the issue is that `_init` with parameters is actually not allowed for instanceable nodes, as when you instance them as PackedScene or via the editor, there's no way to specify arguments.

But that error message should indeed be made clearer, and/or `_init` with arguments should throw a compilation error (but then the issue is that `_init` does support arguments for non-instanceable classes, so it's a bit tricky).

**Comment by profan** [comment]
This is still an issue certainly (3.0.6 and https://github.com/godotengine/godot/commit/8704b77) , and with a scene where the root has a script with a non-zero arg `_init` it will just instantly close the game.

Just got off helping someone who spent quite a long time trying to decipher the somewhat inscrutable error, where if anyone code followed it (like trying to access a member of the instanced scene), what you'll see first is just the membe error, and appear to instance a scene of the node type the scene specified, script ending up as null and access to members failing.

Only after removing any member access to "nonexistent" members, you would see only the real error as above.

Example:
```gdscript
# SomeScript.gd:
var contents
func _init(a, b, c):
    pass

# SomeScene.tscn.. (Imagine it's a Node2D with SomeScene.gd attached)

# SomeGame.gd:
const SomeScene = preload("res://SomeScene.tscn")

func do_things():
    var inst = SomeScene.instance()
    inst.contents = "hello, hello??"`
```

and running this you will be faced with something like this in 3.0.6:
![notgreat](https://user-images.githubusercontent.com/2816340/45843317-8eaae380-bd1f-11e8-90fc-d20c2c93b3ae.png)

while in the version of master I have, you will not get _any_ error, it will merely close out.

**Comment by nightblade9** [comment]
I'm the person who @profan helped with this issue. Just to throw in my two cents, if possible, I would really like this to throw a compilation error with a proper message indicating that you shouldn't use `_init` here for these reasons.

**Comment by Torguen** [comment]
I have this error with v3.5.beta1.official [b9b23d222]

**Comment by celestialaxis** [comment]
This cases to popup when using resource method of saving. (And maybe the same using .store_var() because first I encounter this cases when using .store_var() and I switch to ResrouceSaving in hope to be different)
[https://discordapp.com/channels/212250894228652034/1039776198437376030](url)

**Comment by sancho2** [comment]
I just ran into this issue in version v3.5.1.stable.official [6fed1ffa3]      
Except that there is no error message at all.     
In fact it seems the _init() function with a parameter prevents _ready() from running.  
In the following code nothing prints if _init() has a parameter and there is no error message.   
If _init() has no params. then both "init" and "ready" get printed. 

```
extends Node2D

#func _init()->void: 		# init with no parameter prints "init" and "ready"
	#print("init")

func _init(x:int) -> void:			# has parameter and does not output
	print("init")

func _ready() -> void:
	print("ready")
```

**Comment by HolonProduction** [comment]
The error message for this issue was improved in #86999.
